openfisca_web_api/loader/entities.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals, print_function, division, absolute_import
+from openfisca_core.commons import to_unicode
+
+
+def build_entities(tax_benefit_system):
+    entities = {
+        entity.key: build_entity(entity)
+        for entity in tax_benefit_system.entities
+        }
+    return entities
+
+
+def build_entity(entity):
+
+    entity_formated = {
+        'plural': entity.plural,
+        'description': to_unicode(entity.doc)
+        }
+    if hasattr(entity, 'roles'):
```

Comment:
We usually use:

 ```py
if entity.is_person:
```

Reply:
Fair point. 

We have never used a group entity without roles, and right now it would trigger an ugly error, but you're right, there could be use cases.

However, even if we had a group entity without roles, it would still have an attribute `roles` that would be `[]`, so the code further down would still work. In general, using `hasattr` is pretty rare unless you are doing relatively tricky stuff.